internal/ansible/watches/watches_test.go

```diff
@@ -892,3 +892,49 @@ func TestGetPossibleRolePaths(t *testing.T) {
 		})
 	}
 }
+
+func TestReplaceEnvVars(t *testing.T) { //nolint:gocyclo
+	cwd, err := os.Getwd()
+	if err != nil {
+		t.Fatalf("Unable to get working directory: %v", err)
+	}
+	watchesFile := filepath.Join(cwd, "testdata", "env-vars.yaml")
+	playbookFile := filepath.Join(cwd, "testdata", "playbook.yml")
+
+	os.Setenv("WATCH_PLAYBOOK", playbookFile)
+	defer os.Unsetenv("WATCH_PLAYBOOK")
+	os.Setenv("WATCH_VERSION", "v123")
+	defer os.Unsetenv("WATCH_VERSION")
+	os.Setenv("WATCH_MATCH_LABEL_VAR_NAME", "label123")
+	defer os.Unsetenv("WATCH_MATCH_LABEL_VAR_NAME")
+	os.Setenv("WATCH_MATCH_LABEL_VAR_VALUE", "value123")
+	defer os.Unsetenv("WATCH_MATCH_LABEL_VAR_VALUE")
+	os.Setenv("WATCH_MATCH_EXPRESSIONS_KEY", "key123")
+	defer os.Unsetenv("WATCH_MATCH_EXPRESSIONS_KEY")
```

Comment:
I'd use [`t.Setenv(...)`](https://pkg.go.dev/testing#T.Setenv) here. It will handle the cleanup and reset these to previous values if for some reason they already exist on someone's machine.

Reply:
k.. I can do that. I was just trying to stay consistent with some of the test code I already saw in here. e.g.

line 486ff:
```
os.Setenv("WORKER_MAXCONCURRENTRECONCILESENV_APP_EXAMPLE_COM", "4")
defer os.Unsetenv("WORKER_MAXCONCURRENTRECONCILESENV_APP_EXAMPLE_COM")
os.Setenv("ANSIBLE_VERBOSITY_ANSIBLEVERBOSITYENV_APP_EXAMPLE_COM", "4")
defer os.Unsetenv("ANSIBLE_VERBOSITY_ANSIBLEVERBOSITYENV_APP_EXAMPLE_COM")
```

However, using t.Setenv is a nicer way to do it.

Reply:
this is done.